New Script: Apache Tika

andygrunwald opened the pull request:
## ✍️ Description

Introducing Apache Tika (https://tika.apache.org/) as a new LXC container.

Paperless-NGX can use Apache Tika for additional OCR on Office files (like OpenOffice, MS Office, etc.).

- - -
- Related Discussion: https://github.com/community-scripts/ProxmoxVE/discussions/499 and https://github.com/community-scripts/ProxmoxVE/discussions/1112
- - - 


## ✅ Prerequisites

- [X] Self-review performed (I have reviewed my code to ensure it follows established patterns and conventions.)  
- [] Testing performed (I have thoroughly tested my changes and verified expected functionality.)

## 🛠️ Type of Change

- [] Bug fix (non-breaking change that resolves an issue)  
- [] New feature (non-breaking change that adds functionality)  
- [] Breaking change (fix or feature that would cause existing functionality to change unexpectedly)  
- [X] New script (a fully functional and thoroughly tested script or set of scripts)  

---
## 📋 Additional Information (optional)

### Version detection

Right now, the version of Tika is hardcoded as there is no automatic detection of it yet.
My question would be:

> ❓ Is this fine to get started or is an automatic detection required? If the latter, can you provide some hints on your favorable implementation to avoid ping-pong?

The upgradable functionality depends on the version detection. I can provide the functionality once the version detection works.

### Testing

The LXC Container setup is working as expected already.

Before merge, I would like to extend my testing of this setup by integrating it into Paperless to see if everything is complete.

Once done, I will revert https://github.com/community-scripts/ProxmoxVE/commit/f1c5d8720696cdbfde9471abfff07e4b7b71bc6d.

### Resource usage

The entered resources are a guess right now.
I don't have experience if this is over- or undersized.

### Configuration

There is the possibility to configure Apache Tika.
This script is not offering a default config file, as Apache Tika is shipped with sane defaults that should be a good choice for the majority of users.

michelroegl-brunner commented:
You have build.func and install.func in there...

andygrunwald commented:
@michelroegl-brunner This is correct. However, i stated in the PR description a few things like

> Testing
> The LXC Container setup is working as expected already.
>
>Before merge, I would like to extend my testing of this setup by integrating it into Paperless to see if everything is complete.
>
> Once done, I will revert https://github.com/community-scripts/ProxmoxVE/commit/f1c5d8720696cdbfde9471abfff07e4b7b71bc6d.

On top of this, a few questions, e.g. the hardcoded version.
See

> Version detection
> Right now, the version of Tika is hardcoded as there is no automatic detection of it yet.
My question would be:
> 
> ❓ Is this fine to get started or is an automatic detection required? If the latter, can you provide some hints on your favorable implementation to avoid ping-pong?
> 
> The upgradable functionality depends on the version detection. I can provide the functionality once the version detection works.

If this is not the correct way to request feedback/guidance from you, please let me know what the correct way is.

MickLesk commented:
why not:
RELEASE="$(wget -qO- https://dlcdn.apache.org/tika/ | grep -oP '(?<=href=")[0-9]+\.[0-9]+\.[0-9]+(?=/")' | sort -V | tail -n1)"
wget -q "https://dlcdn.apache.org/tika/${RELEASE}/tika-server-standard-${RELEASE}.jar"

bvdberg01 commented on install/apache-tika-install.sh:
```diff
@@ -0,0 +1,80 @@
+  cabextract
+msg_ok "Installed Dependencies"
+
+msg_info "Setup Java"
+$STD apt-get install -y \
+  default-jdk
+msg_ok "Setup Java"
```
Can this be moved to install dependencies?

MickLesk replied:
Can stay there, but would take the opensource name.... “Setup OpenJDK”

andygrunwald replied:
Switched to `openjdk-17-jre-headless` in https://github.com/community-scripts/ProxmoxVE/pull/2079/commits/17544c923514b7a1ca9688a93a0e933898f10d1a

This is in line with the official dockerfile: https://github.com/apache/tika-docker/blob/e8819e3e2a8d29e803ecdc6b0c6169ed19c5b1c3/full/Dockerfile#L46C10-L46C33

bvdberg01 commented on install/apache-tika-install.sh:
```diff
@@ -0,0 +1,80 @@
+update_os
+
+msg_info "Installing Dependencies"
+# See also https://github.com/apache/tika-docker/blob/master/full/Dockerfile
```
Remove comment

andygrunwald replied:
Fixed in https://github.com/community-scripts/ProxmoxVE/pull/2079/commits/5d49a7d6b94cdfc6c3bd5f3b5fed2a08eb8268ed

bvdberg01 commented on install/apache-tika-install.sh:
```diff
@@ -0,0 +1,80 @@
+[Install]
+WantedBy=multi-user.target
+EOF
+
```
Remove empty line

andygrunwald replied:
Fixed in https://github.com/community-scripts/ProxmoxVE/pull/2079/commits/1c72574d536da533c4e187eed6f1e1188591cfb0

MickLesk commented on install/apache-tika-install.sh:
```diff
@@ -0,0 +1,80 @@
+  software-properties-common \
+  gdal-bin \
+  tesseract-ocr \
+  tesseract-ocr-eng \
```
Single Line for all langs? ocr{eng,deu...}

andygrunwald commented on json/apache-tika.json:
```diff
@@ -0,0 +1,34 @@
+    ],
+    "date_created": "2025-02-05",
+    "type": "ct",
+    "updateable": false,
```
Fixed in https://github.com/community-scripts/ProxmoxVE/pull/2079/commits/9a0605b6bd66552f60c0cbce190788381dd1b60c

andygrunwald commented on ct/apache-tika.sh:
```diff
@@ -0,0 +1,67 @@
+    msg_info "Starting ${APP}"
+    systemctl start apache-tika
+    msg_ok "Started ${APP}"
+    msg_ok "Updated Successfully"
```
Fixed in https://github.com/community-scripts/ProxmoxVE/pull/2079/commits/d107f3d0908621b62de9558e2d60d22f459be716

andygrunwald commented:
@michelroegl-brunner @MickLesk @bvdberg01 I fixed (nearly) all open comments and reverted my URL-changes (to make it work with my fork).

Additionally, I tested the installation in the default way: Successful

<img width="1089" alt="Screenshot 2025-02-06 at 17 52 17" src="https://github.com/user-attachments/assets/803568ef-410e-4720-8401-83aef06b8933" />

I downgraded manually to Apache Tika v2.9.2 and tested the update functionality: Successful

<img width="401" alt="Screenshot 2025-02-06 at 17 56 32" src="https://github.com/user-attachments/assets/624885b7-479f-4f0d-a203-408d6aac61f4" />

I would like to request a new / final review from your side. Thank you.

andygrunwald commented on ct/apache-tika.sh:
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+#source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
+source <(curl -s https://raw.githubusercontent.com/andygrunwald/ProxmoxVE/refs/heads/new-script-apache-tika/misc/build.func)
```
Please see the PR description. This was done during testing, as I requested guidance.
This has been reverted a few minutes ago, see https://github.com/community-scripts/ProxmoxVE/pull/2079/commits/43bff08c5689afb76d02993bb108d6504c8fdeb6

tremor021 reviewed (approved):
LGTM
